### ISSUE TITLE
docker run error "cannot import name 'soft_unicode' from 'markupsafe'"

### ISSUE THREAD

**Issue by ming54864** [mitre/caldera#2496]
git clone https://github.com/mitre/caldera.git --recursive --branch 4.0.0-beta
docker build . -t caldera:server  
docker run -p 7010:7010 -p 7011:7011/udp -p 7012:7012 -p 8888:8888 caldera:server

error:
Traceback (most recent call last):
  File "server.py", line 8, in <module>
    import aiohttp_apispec
  File "/usr/local/lib/python3.8/dist-packages/aiohttp_apispec/__init__.py", line 1, in <module>
    from .aiohttp_apispec import AiohttpApiSpec, setup_aiohttp_apispec
  File "/usr/local/lib/python3.8/dist-packages/aiohttp_apispec/aiohttp_apispec.py", line 10, in <module>
    from jinja2 import Template
  File "/usr/local/lib/python3.8/dist-packages/jinja2/__init__.py", line 12, in <module>
    from .environment import Environment
  File "/usr/local/lib/python3.8/dist-packages/jinja2/environment.py", line 25, in <module>
    from .defaults import BLOCK_END_STRING
  File "/usr/local/lib/python3.8/dist-packages/jinja2/defaults.py", line 3, in <module>
    from .filters import FILTERS as DEFAULT_FILTERS  # noqa: F401
  File "/usr/local/lib/python3.8/dist-packages/jinja2/filters.py", line 13, in <module>
    from markupsafe import soft_unicode
ImportError: cannot import name 'soft_unicode' from 'markupsafe' (/usr/local/lib/python3.8/dist-packages/markupsafe/__init__.py)

**Comment by b4b857f6ee** [comment]
I got this one too...

**Comment by b4b857f6ee** [comment]
@ming54864 i found the solution : https://github.com/aws/aws-sam-cli/issues/3661

just run `pip3 install markupsafe==2.0.1`

**Comment by ArtificialErmine** [comment]
Hey @ming54864, @b4b857f6ee - this is indeed an issue with the markupsafe library and how certain third party libraries we use didn't pin specific versions of other third party libraries which then got updated. Unfortunately, we can't just update the first collection of third party libraries just yet due to conflicting dependencies, but we're hopeful they'll sort that out soon. 

In the meantime, solving this issue can be done either by using the approach outlined (`pip3 install markupsafe==2.0.1`, or by updating to the the current master of Caldera, where we've fixed the problem manually in our requirements.txt file.
